src/service/CredentialService.ts

```diff
@@ -139,6 +215,23 @@ CredentialService.Router.post("/token", async (req: Request, res: Response) => {
   res.json(result)
 })
 
+
+CredentialService.Router.post("/personal_access_token", async (req: Request, res: Response) => {
```

Comment:
Is this the only new API endpoint we are adding? Is there one for revoking the token? Or is that handled by removing a token from the array and calling `Credential.update`?

Reply:
Oh, actually, I see here that `grant_type` can generate and revoke tokens in this single call. 

Could we rename this endpoint to `PUT /credential/{id}/token` →generate token & `DELETE /credential/{id}/token/{token_string}` →revoke tokens? Essentially just splitting up this one method into two. (id in this case would be access_key.)

Would it make sense too to make `expiresAt` be a UNIX epoch duration or timestamp (in milliseconds)? We use those to represent date/time across the API so we want to avoid date-strings. I am not sure if a hard timestamp is better or a duration is better in this case.

Reply:
I feel that having it in ms makes sense, but if you prefer to have it as duration it's also good :), however in that case I would rename the property to validDuration or something like that to be more specific.
One question that came up.
Do we want to be able to update the expire date?
If we don't want to do that @falmeida-orangeloops pointed out that we should use the `expiresIn` property (which takes duration FE "1 week") which is verified when by JWT.
In this case the user should regenerate a new personal access token when the token expires.
Having it in the db allows us to update the expiry date without modifying the token.
In that case we should also add an update endpoint, to change the expiresAt.

Reply:
Let's go with `created` and `expiry` for the field names. I think UNIX epoch timestamp makes sense instead of a duration which can be tricker to work with. 

The expiry should not be updatable, and we don't want an API endpoint for that. I think we can encode it into the JWT too, but it may help to also keep it in the PAT object so we can display that in the UI (i.e. "token expired" or "token expires in 3 days" or something).